### PR TITLE
Add support for the Illumos platform

### PR DESCRIPTION
**Pull request by bardo**
Hello!

I run SmartOS, a distribution based on the Illumos OS, and recently one of the projects I rely on failed to build due to a dependency on this package. With this PR I am adding support for this platform, with a couple of caveats:
1. I am an extreme beginner with Rust, so my code builds, but it may very likely be incorrect. Code style is likely lacking.
2. I cannot guarantee that my implementation is complete. I looked around for all the `target_os` mentions and I think I covered them all. This may or may not be true.

As for the implementation details:
* Illumos, being forked from OpenSolaris, [follows its original definition](https://github.com/illumos/illumos-gate/blob/85fe9a719d390fcef3899d53f152b562c75d354f/usr/src/uts/common/sys/utsname.h#L52-L58) for the `utsname` struct, which is the same of *BSD.
* There is no systemd, so no `/etc/machine-info`, and for `devicename()` I am reading `/etc/nodename`. This usually matches the hostname, but not necessarily.

If any changes are needed to this PR kindly drop a comment, or feel free to edit it yourself if you prefer.

Thank you!

### CODE REVIEW

File: src/unix.rs [hunk 1]
```diff
@@ -348,6 +348,18 @@ pub(crate) fn devicename_os() -> OsString {
...
 }
 
+#[cfg(target_os = "illumos")]
+pub(crate) fn devicename() -> String {
+    let mut nodename = String::new();
+
+    if let Ok(program) = std::fs::read_to_string("/etc/nodename") {
+        let program = program.into_bytes();
+        nodename.push_str(&String::from_utf8_lossy(&program));
+        nodename.pop();
```
Comment (AldaronLau) [inline src/unix.rs]: Can you add a comment saying which character is being popped here?  Is it a newline, null byte, etc.?

**Review by AldaronLau — APPROVED** [review]
@bardo Thanks for opening this PR!  Don't worry about the CI not passing (it's not because of this PR).  I just have one minor comment, otherwise looks good to merge.

File: src/unix.rs (hunk 1 above)
Comment (bardo) [reply src/unix.rs]: Updated!

**Comment by AldaronLau** [comment]
@bardo Just released `whoami` 1.4.0 with this patch!

**Pull request merged**